docs/pre-commit.md

```diff
@@ -0,0 +1,39 @@
+# Using pre-commit locally
+
+### Install and setup pre-commit
+- Install:
+  - Using pip: `pip install pre-commit`
+  - Using Homebrew: `brew install pre-commit`
+  - see docs [Click](https://pre-commit.com/#install)
+
+- Setup
+  - run `pre-commit install` to set up the git hook script
+  - see docs [Click](https://pre-commit.com/#install)
+
+### Things to keep in mind regarding the git authoring lifecycle (HEAD, Staged, Unstaged) and pre-commit hooks
+
+Running hooks on unstaged changes can lead to both false-positives and false-negatives during committing. pre-commit only runs on the staged contents of files by temporarily saving the contents of your files at commit time and stashing the unstaged changes while running hooks.
```

Comment:
```suggestion
Running hooks on unstaged changes can lead to both false-positives and false-negatives during committing, so to avoid this, pre-commit only runs on the staged contents of files by temporarily saving the contents of your files at commit time and stashing the unstaged changes while running hooks.
```

Reply:
This would make it clearer to me, but only a suggestion